### PR TITLE
Convert mod_muc_light_cache to gen_hook

### PR DESCRIPTION
**Pull request by kamilwaz**
Convert mod_muc_light_cache to gen_hook.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/147323) / small_tests / be1026d6460d957d45250930329cfb7aba0e9ab7
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3828/147323/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3828/147323/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-10-26_13.39.47/lib.mongooseim.logs/run.2022-10-26_13.39.48/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/147386) / small_tests / 20fed8856ee05be15ccb432802098fb0fef6c12c
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3828/147386/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3828/147386/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-10-27_08.14.16/lib.mongooseim.logs/run.2022-10-27_08.14.17/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/147464) / small_tests / a1b9f51cea0602e89bb2ff755971171262f806fb
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3828/147464/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3828/147464/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-10-27_10.02.08/lib.mongooseim.logs/run.2022-10-27_10.02.09/suite.log.html)

File: src/muc_light/mod_muc_light.erl [hunk 1]
```diff
@@ -378,66 +376,78 @@ make_err(From, To, El, Acc, Reason) ->
...
+    Extra :: map().
+prevent_service_unavailable(Acc, #{packet := Packet}, _Extra) ->
     case xml:get_tag_attr_s(<<"type">>, Packet) of
         <<"groupchat">> -> {stop, Acc};
-        _Type -> Acc
+        _Type -> {ok, Acc}
     end.
 
--spec disco_local_items(mongoose_disco:item_acc()) -> mongoose_disco:item_acc().
+-spec disco_local_items(Acc, Params, Extra) -> {ok | stop, Acc} when
```
Comment (pawlooss1) [inline src/muc_light/mod_muc_light.erl:389]: This handler will actually never return `{stop, Acc}`. Maybe we should change the return type of this spec to just `{ok, Acc}`? If you agree then please change it for other handlers which don't use `stop` atom.

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/147516) / small_tests / c3d607d26e07ed3e09717be4833ea0d206427738
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3828/147516/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3828/147516/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-10-27_11.50.21/lib.mongooseim.logs/run.2022-10-27_11.50.21/suite.log.html)

**Review by pawlooss1 — APPROVED** [review]
👌

**Pull request merged**